File: PHP/88. Merge Sorted Array.php
<?php

/**
 * @noinspection PhpIllegalPsrClassPathInspection
 * @noinspection PhpMultipleClassDeclarationsInspection
 * @noinspection PhpUnused
 */
class Solution
{
    /**
     * @param  int[]  $nums1
     * @param  int[]  $nums2
     */
    public function merge(array &$nums1, int $m, array $nums2, int $n): void
    {
        $m--;
        $n--;

        for ($i = ($m + $n + 1); $i >= 0; $i--) {
            if ($m < 0) {
                $nums1[$i] = $nums2[$n];
                $n--;

                continue;
            }

            if ($n < 0) {
                $nums1[$i] = $nums1[$m];
                $m--;

                continue;
            }

            if ($nums2[$n] > $nums1[$m]) {
                $nums1[$i] = $nums2[$n];
                $n--;
            } else {
                $nums1[$i] = $nums1[$m];
                $m--;
            }
        }
    }
}

File: Python/88. Merge Sorted Array.py
from typing import List


# noinspection PyMethodMayBeStatic
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        m -= 1
        n -= 1

        for i in range(m + n + 1, -1, -1):
            if m < 0:
                nums1[i] = nums2[n]
                n -= 1
                continue

            if n < 0:
                nums1[i] = nums1[m]
                m -= 1
                continue

            if nums2[n] > nums1[m]:
                nums1[i] = nums2[n]
                n -= 1
            else:
                nums1[i] = nums1[m]
                m -= 1
